Migration to PyVO for MAST TAP notebooks

tsdower opened the pull request:
As the ESA TAP/TAPPlus library is being superceded by increased support for PyVO, we are moving MAST's notebooks previously using the TAP+ client functionality to PyVO's TAP client support. New notebooks from this point will be designed with PyVO in mind. 

Development outside these notebooks is still required to clear spurious warnings, particularly regarding units, which also produced similar warnings in the TAP+ system. Once these are handled, the warning 'ignore' can be removed.

tsdower committed:
Merge pull request #7 from spacetelescope/master

updating personal fork from master

tsdower committed:
Update HSCTAP to PyVO

Update HSC(v3) TAP examples and text to use PyVO

tsdower committed:
HSC_TAP: show description for ALL user tables

tsdower committed:
update existing TESS (CAOM) and PanSTARRS notebooks to pyvo

theresadower commented:
@ceb8: Spoke to @eteq, can you review this as you're more familiar with ASB materials?
In general, we noted: 
 * async pyvo is slower than the async TAP+ calls we had before, but not slower enough to fail CI
 * there are ignored warnings from pyvo, particularly about units. some of these are the same as we had from TAP+. Some will need addressed in pyvo.
 * there will be resources dedicated to improving both of these as part of the general pyvo development.

ceb8 commented on notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb:
```diff
@@ -38,7 +38,7 @@
     "Table Access Protocol (TAP) services allow more direct and flexible access to astronomical data than the simpler types of IVOA standard data services. Queries are built with the SQL-like Astronomical Data Query Language (ADQL), and can include geographic / spatial queries as well as filtering on other characteristics of the data. This also allows the user fine-grained control over the return ...
     "\n",
-    "For this example, we'll be using the astroquery TAP/TAP+ client, which was developed by the ESAC Space Data Centre for working with the GAIA catalog, but is interoperable with other valid TAP services, including those at MAST. As an astroquery project, TAP+ documentation is available at ReadTheDocs: http://astroquery.readthedocs.io/en/latest/utils/tap.html\n",
+    "For this example, we'll be using the astropy affiliated PyVO client, which is interoperable with other valid TAP services, including those at MAST. PyVO documentation is available at ReadTheDocs: https://readthedocs.org/projects/pyvo/\n",
```
https://readthedocs.org/projects/pyvo/ -> https://pyvo.readthedocs.io

ceb8 commented on notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb:
```diff
@@ -38,7 +38,7 @@
-    "For this example, we'll be using the astroquery TAP/TAP+ client, which was developed by the ESAC Space Data Centre for working with the GAIA catalog, but is interoperable with other valid TAP services, including those at MAST. As an astroquery project, TAP+ documentation is available at ReadTheDocs: http://astroquery.readthedocs.io/en/latest/utils/tap.html\n",
+    "For this example, we'll be using the astropy affiliated PyVO client, which is interoperable with other valid TAP services, including those at MAST. PyVO documentation is available at ReadTheDocs: https://readthedocs.org/projects/pyvo/\n",
     "\n",
     "We'll be using TAP+ to call the TAP service at MAST serving PanSTARRS 1 Data Release 2, now with individual detection information. The schema is described within the service, and we'll show how to inspect it. The schema is also the same as the one available via the CasJobs interface, with an additional view added for the most common positional queries. CasJobs has its own copy of the schema  ...
```
TAP+ -> PyVO

ceb8 commented on notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb:
```diff
@@ -109,7 +114,8 @@
    "metadata": {},
```
Comment for line 103/108 (can't comment on that line):
TapPlus -> PyVO

ceb8 commented on notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb:
```diff
@@ -199,8 +208,8 @@
     "\"\"\".format(ra,dec,radius)\n",
```
Comment for line 188/197:
`Mast._resolve_object(objname)` -> `Mast.resolve_object(objname)`
(Astroquery changed the name of this function)

theresadower replied:
Updated this. The non-TAP HSC notebook may need updated as well, @rlwastro wrote that one.

ceb8 replied:
I can check and do a quick update PR on that one if needed.

ceb8 commented on notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb:
```diff
@@ -491,8 +500,8 @@
     "\"\"\".format(ra,dec,radius)\n",
```
Comment for line 481/490:
`Mast._resolve_object(objname)` -> `Mast.resolve_object(objname)`

ceb8 commented on notebooks/MAST/PanSTARRS/PS1_DR2_TAP/PS1_DR2_TAP.ipynb:
```diff
@@ -635,8 +644,8 @@
     "\"\"\".format(ra,dec,radius)\n",
```
Comment for line 625/634:
`Mast._resolve_object(objname)` -> `Mast.resolve_object(objname)`

ceb8 commented on notebooks/MAST/TESS/beginner_tess_tap_search/beginner_tess_tap_search.ipynb:
```diff
@@ -41,10 +41,9 @@
     "\n",
-    "We'll be using TAP+ to call the CAOM Catalog TAP service at MAST and filter the results for TESS-related information. The schema for this catalog is an IVOA standard, and is also described within the service itself.\n",
-    "\n"
+    "We'll be using TAP+ to call the CAOM Catalog TAP service at MAST and filter the results for TESS-related information. The schema for this catalog is an IVOA standard, and is also described within the service itself."
```
TAP+ -> PyVO

ceb8 commented on notebooks/MAST/TESS/beginner_tess_tap_search/beginner_tess_tap_search.ipynb:
```diff
@@ -110,7 +115,29 @@
    "metadata": {},
```
Comment for line 104/109: 
TapPlus -> PyVO

ceb8 reviewed (changes requested):
There are a number of TESS lc fits files in the TESS notebook directory. Does the notebook require them or are they files produced by the notebook that accidentally got checked in?

Overall looks great, just a few comments, mostly places where you still refer to tap+. Also, I recommend updating your version of astroquery.

ceb8 commented:
Oh wait, ignore what I said about the extra files. I was looking at my local directory when I thought I was looking at github. Sorry! 😛

tsdower committed:
documentation updates and Mast.resolve_object

Further documentation updates to match PyVO changes.
update Mast resolve_object calls.

theresadower commented on notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb:
```diff
@@ -42,9 +42,9 @@
     "Table Access Protocol (TAP) services allow more direct and flexible access to astronomical data than the simpler types of IVOA standard data services. Queries are built with the SQL-like Astronomical Data Query Language (ADQL), and can include geographic / spatial queries as well as filtering on other characteristics of the data. This also allows the user fine-grained control over the return ...
     "\n",
-    "For this example, we'll be using the astroquery TAP/TAP+ client, which was developed by the ESAC Space Data Centre for working with the GAIA catalog, but is interoperable with other valid TAP services, including those at MAST. As an astroquery project, TAP+ documentation is available at ReadTheDocs: http://astroquery.readthedocs.io/en/latest/utils/tap.html\n",
+    "For this example, we'll be using the astropy affiliated PyVO client, which is interoperable with other valid TAP services, including those at MAST. PyVO documentation is available at ReadTheDocs: https://readthedocs.org/projects/pyvo/\n",
```
Updated in this and the other notebooks, thanks.

theresadower commented on notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb:
```diff
@@ -42,9 +42,9 @@
    "source": [
     "Table Access Protocol (TAP) services allow more direct and flexible access to astronomical data than the simpler types of IVOA standard data services. Queries are built with the SQL-like Astronomical Data Query Language (ADQL), and can include geographic / spatial queries as well as filtering on other characteristics of the data. This also allows the user fine-grained control over the return ...
```
Fixed

theresadower commented on notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb:
```diff
@@ -109,7 +111,7 @@
    "metadata": {},
```
Whoops, fixed this in this and the other notebooks, thanks.
